torchmultimodal/utils/attention.py

```diff
@@ -43,6 +45,49 @@ def get_extended_attention_mask(attention_mask: Tensor) -> Tensor:
     return extended_attention_mask
 
 
+def get_extended_attention_mask_for_decoder(attention_mask: Tensor) -> Tensor:
```

Comment:
This seems to be a very specific use case where you want to derive the causal mask from the padding mask.
Instead, one can also create the causal mask "from first principle" by knowing the sequence lengths and other dimensions, right?
Also, I would look at how we can unify with `get_causal_attention_mask`.